accepted/future-releases/enhanced-enums/feature-specification.md

```diff
@@ -68,128 +66,175 @@ must redirect to a generative constructor._
+We change the publicly visible API of the  `Enum` class to be declared as:
 
-The semantics of such an enum declaration is defined as being equivalent to *rewriting into a corresponding class declaration* as follows:
-
-- Declare a class with the same name and type parameters as the `enum` declaration.
-
-- Add `extends Enum`, where `Enum` refers to the type declared in `dart:core`.
+```dart
+class Enum {
+  // No default constructor (however we enforce that).
+  external int get index;
+  external String toString();
+}
+```
 
-- Further add the mixins and interfaces of the `enum` declaration.
+We intend to (at least pretend to) let `enum` classes extend `Enum`, and let mixins and members access the default `index` and `toString()` through `super.`. _(In practice, we may use a different class implementing `Enum` as the superclass, but for checking the validity of `super.index`/`super.toString()`, we analyze against `Enum` itself, so it must have non-abstract implementations.)_
+
+This all makes it look as if `Enum` would be a valid superclass for the mixin applications and methods of the enhanced `enum` class.
+
+The semantics of such an enum declaration, *E*, is defined as introducing a (semantic) *class*, *C*, just like a similar `class` declaration.
+
+* **Name**: The name of the class *C* and its implicit interface is the name of the `enum` declaration.
+
+* **Superclass**: The superclass of *C* is (seemingly) the `Enum` class, with the mixins declared by *E* applied. _(In practice, the superclass may be another system provided class with implements `Enum`.)_
+
+  * If *E* is declared as `enum Name with Mixin1, Mixin2 …` then the superclass of *C* is the mixin application `Enum with Mixin1, Mixin2`.
+
+  It’s a **compile-time error** if such a mixin application introduces any instance variables. _We need to be able to call an implementation specific superclass `const` constructor of `Enum`, and a mixin application of a mixin with a field does not make its forwarding constructor `const`. Currently that’s the only restriction, but if we add further restrictions on mixin applications having `const` forwarding constructors, those should also apply here._
+
+* **Superinterfaces**: The immediate superinterfaces of *C* are the interface of the superclass and the interfaces declared by *E*.
+  * If `E` is declares as `enum Name with Mixin1, Mixin2 implements Type1, Type2 { … }` then the immediate superinterfaces of *C* are the interfaces of `Name with Mixin1, Mixin2`, `Type1` and `Type2`.
+
+- **Declared members**: For each member declaration of the `enum` declaration *E*, the same member is added to the class *C*. This includes constructors (which must be `const` generative or non-`const` factory constructors.)
+
+- **Default constructor**: If no generative constructors were declared, and no unnamed factory constructor was added,
+  a default generative constructor is added:
+
+  ```dart
+  const Name();
+  ```
```

Comment:
So this constructor is assumed to be transformed later on, to accept the two "extra" arguments?